docs/ADR-71-upgradable-collections.md

```diff
@@ -0,0 +1,53 @@
+# Upgradable Collections
+
+## Context
+
+![MinimalProxy](./resources/ADR-71/collection-creation-old-flow.png)
+
+This diagram shows the current flow regarding publishing collections on the blockchain. 
+
+The [CollectionManager](https://polygonscan.com/address/0x9D32AaC179153A991e832550d9F96441Ea27763A) has a `createCollection` external function called by the creator of a collection which receives both the address of the [Forwarder](https://polygonscan.com/address/0xBF6755A83C0dCDBB2933A96EA778E00b717d7004) and the [CollectionFactory](https://polygonscan.com/address/0xB549B2442b2BD0a53795BC5cDcBFE0cAF7ACA9f8)
+
+Inside this function, the Forwarder calls another `createCollection` function the CollectionFactory contract.
+
+This creates a new collection in the blockchain in the form of a [MinimalProxy](https://eips.ethereum.org/EIPS/eip-1167), which delegates all calls made to it to the [CollectionImplementation](https://polygonscan.com/address/0x006080C6061C4aF79b39Da0842a3a22A7b3f185e) contract containing the logic of how collections should behave.
+
+Without this pattern, every time a collection is created, a new CollectionImplementation contract would have to be deployed. And this contract is BIG, costing a lot of gas in the process.
+
+Thanks to how the EVM handles storage, These proxies can delegate calls to a contract containing the behavior, but updating only their own state.
+
+## Problem
+
+Let's say a vulnerability is found in the CollectionImplementation contract, or a new magical feature for collections is voted by the community to be added.
+
+With this current implementation, sadly, there is absolutely nothing we can do. The proxies delegate calls to the implementation contract directly, and that does not allow any kind of modification or upgrade to the collection behavior.
```

Comment:
```suggestion
With this current implementation, sadly, there is absolutely nothing we can do. The proxies delegate calls to the implementation contract directly, and that does not allow any kind of modification or upgrade to the collection behavior because they are not upgradable proxies.
```